torch/_refs/__init__.py

```diff
@@ -3285,11 +3299,27 @@ def new_zeros(
     device: Optional[torch.device] = None,
     pin_memory: bool = False,
 ) -> TensorLikeType:
-    r = a.new_empty(
-        size, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory
+    dtype = a.dtype if dtype is None else dtype
+    layout = a.layout if layout is None else layout
```

Comment:
I think the documentation for `new_zeros` it out-of-date:

https://pytorch.org/docs/master/generated/torch.Tensor.new_zeros.html?highlight=new_zeros#torch.Tensor.new_zeros

It doesn't even refer to layout or pinning memory

Reply:
Tracked with https://github.com/pytorch/pytorch/issues/84986